src/Microsoft.AspNet.HttpOverrides/HttpMethodOverrideExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.HttpOverrides;
+
+namespace Microsoft.AspNet.Builder
+{
+    public static class HttpMethodOverrideExtensions
+    {
+        /// <summary>
+        /// Allows the user to override the default method type of a request
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseHttpMethodOverrideMiddleware(this IApplicationBuilder builder)
+        {
+            return builder.Use(next => new HttpMethodOverrideMiddleware(next).Invoke);
+        }
+
+        /// <summary>
+        /// Allows the user to override the default method type of a request
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="inputName">Denotes the element that contains the name of the resulting method type</param>
```

Comment:
This parameter seems like it's most specifically `formFieldName`, is that the case?